### PR TITLE
Add header-with-button 'component'

### PR DESCRIPTION
**Pull request by spencer**
This is an extraction from a broader piece of 'design'.

A basic 'component' (no JS required) to allow headings to have buttons aligned right. After conversation with @benbarnett we decided to do this with table / table cell styling, as it's been used elsewhere.

### CODE REVIEW

File: app/assets/stylesheets/_heading_with_button.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.heading-with-button {
+  display: table;
+  width: 100%;
+}
+.heading-with-button__header { display: table-cell}
```
Comment (benbarnett) [inline app/assets/stylesheets/_heading_with_button.scss]: Please drop onto block and add semicolon

File: app/assets/stylesheets/_heading_with_button.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+.heading-with-button__button { 
+  display: table-cell;
+  text-align: right
```
Comment (benbarnett) [inline app/assets/stylesheets/_heading_with_button.scss]: semicolon

**Comment by Guntrisoft** [comment]
:+1: after ben's comments are implemented

**Comment by benbarnett** [comment]
:+1: same

File: app/assets/stylesheets/_heading_with_button.scss (hunk 1 above)
Comment (spencer) [inline app/assets/stylesheets/_heading_with_button.scss]: doing it now, sorry - Caden flagged it and whitespace

**Comment by benbarnett** [comment]
Just noticed as well you're not using the SASS folder structure we have elsewhere. Fine for now but could become unwieldy.

:+1:

**Pull request merged**